test/functional/feature_asset_locks.py

```
@@ -758,6 +764,69 @@ def test_v24_fork(self, node_wallet, node, pubkey):
 
+    def test_asset_locks_v2_pre_v24(self, node_wallet, node, pubkey):
+        self.log.info("Testing asset lock v2 rejection before v24 activation...")
+        assert not softfork_active(node_wallet, 'v24')
+
+        self.mempool_size = node_wallet.getmempoolinfo()['size']
+
+        coins = node_wallet.listunspent(query_options={'minimumAmount': 1})
+        coin = coins.pop()
+        lock_tx = self.create_assetlock(coin, COIN, pubkey, version=2)
+        self.check_mempool_result(tx=lock_tx,
+            result_expected={'allowed': False, 'reject-reason': 'bad-assetlocktx-version-2'})
```

Flake8 (7.3.0)

[error] continuation line under-indented for visual indent

(E128)